vulnerabilities/data_source.py

```diff
@@ -88,14 +86,17 @@ class Advisory:
 
     summary: str
     vulnerability_id: Optional[str] = None
-    affected_packages: List[AffectedPackage] = dataclasses.field(default_factory=list)
+    affected_package_urls: Iterable[PackageURL] = dataclasses.field(default_factory=list)
+    fixed_package_urls: Iterable[PackageURL] = dataclasses.field(default_factory=list)
```

Comment:
We were not capturing version ranges earlier as well. Ranges were deflated into raw versions. As on https://github.com/nexB/vulnerablecode/blob/b6ef11d8c3552618f3ca4608ecd59611c6515df4/vulnerabilities/importers/nginx.py#L152-L191

Probably it will make sense to store version ranges _or_ concrete versions depending on the importer data source (as we don't want to tamper with the data source at all) . This will need further thought.

Reply:
New commits address this by using ``AffectedPackage`` class and having a VersionSpecifier